environment/src/main/java/com/sequenceiq/environment/environment/service/freeipa/FreeIpaPollerService.java

```diff
@@ -62,14 +63,16 @@ public void waitForSynchronizeUsers(Long envId, String envCrn) {
 
     public void waitForCcmUpgrade(Long envId, String envCrn) {
         OperationStatus status = freeIpaService.upgradeCcm(envCrn);
-        try {
-            Polling.stopAfterAttempt(upgradeccmAttempt)
-                    .stopIfException(true)
-                    .waitPeriodly(upgradeccmSleeptime, TimeUnit.SECONDS)
-                    .run(() -> freeipaPollerProvider.upgradeCcmPoller(envId, envCrn, status.getOperationId()));
-        } catch (PollerStoppedException e) {
-            LOGGER.info("FreeIPA Upgrade CCM timed out or error happened.", e);
-            throw new FreeIpaOperationFailedException("FreeIPA upgrade of Cluster Connectivity Manager timed out or error happened: " + e.getMessage());
+        if (status.getStatus() != OperationState.COMPLETED) {
+            try {
+                Polling.stopAfterAttempt(upgradeccmAttempt)
+                        .stopIfException(true)
+                        .waitPeriodly(upgradeccmSleeptime, TimeUnit.SECONDS)
+                        .run(() -> freeipaPollerProvider.upgradeCcmPoller(envId, envCrn, status.getOperationId()));
```

Comment:
Sanity check: Does it make sense to poll in case of some failure state (`FAILED`, `REJECTED`, `TIMEDOUT`)? Possibly we should be only polling when `REQUESTED` or `RUNNING`?

Reply:
The poller will exit correctly in the abovementioned states.
COMPLETED is a special case, as for retries, if FreeIPA is already upgraded, then no flow is triggered but a new OperationStatus is returned with COMPLETED state and a freshly generated Operation ID.
This Operation ID does not exist in the database as such, so the subsequent `freeIpaService.getOperationStatus(operationId)` call would throw an error. This is prevented with the condition.